### ISSUE TITLE
Category search breaks on 500 internal server error

### ISSUE THREAD

**Issue by tobixen** [Kozea/Radicale#1280]
I thought I had reported this earlier, but apparently not.

Here is a traceback:

```
 An exception occurred during REPORT request on '/user1/pythoncaldav-test/': Failed to filter item '19970901T130000Z-123403@example.com.ics' from 'user1/pythoncaldav-test': 'list' object has no attribute 'lower'
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/radicale/app/report.py", line 128, in xml_report
    if not all(test_filter(collection_tag, item, filter_)
  File "/usr/lib/python3.10/site-packages/radicale/app/report.py", line 128, in <genexpr>
    if not all(test_filter(collection_tag, item, filter_)
  File "/usr/lib/python3.10/site-packages/radicale/app/report.py", line 251, in test_filter
    return radicale_filter.comp_match(item, filter_[0])
  File "/usr/lib/python3.10/site-packages/radicale/item/filter.py", line 105, in comp_match
    if not comp_match(item, child, level=level + 1):
  File "/usr/lib/python3.10/site-packages/radicale/item/filter.py", line 98, in comp_match
    if not any(prop_match(comp, child, "C")
  File "/usr/lib/python3.10/site-packages/radicale/item/filter.py", line 98, in <genexpr>
    if not any(prop_match(comp, child, "C")
  File "/usr/lib/python3.10/site-packages/radicale/item/filter.py", line 135, in prop_match
    if not text_match(vobject_item, child, name, ns):
  File "/usr/lib/python3.10/site-packages/radicale/item/filter.py", line 471, in text_match
    condition = any(match(child.value) for child in children)
  File "/usr/lib/python3.10/site-packages/radicale/item/filter.py", line 471, in <genexpr>
    condition = any(match(child.value) for child in children)
  File "/usr/lib/python3.10/site-packages/radicale/item/filter.py", line 454, in match
    value = value.lower()
AttributeError: 'list' object has no attribute 'lower'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/radicale/app/__init__.py", line 105, in __call__
    status_text, headers, answers = self._handle_request(environ)
  File "/usr/lib/python3.10/site-packages/radicale/app/__init__.py", line 287, in _handle_request
    status, headers, answer = function(
  File "/usr/lib/python3.10/site-packages/radicale/app/report.py", line 299, in do_REPORT
    status, xml_answer = xml_report(
  File "/usr/lib/python3.10/site-packages/radicale/app/report.py", line 135, in xml_report
    raise RuntimeError("Failed to filter item %r from %r: %s" %
RuntimeError: Failed to filter item '19970901T130000Z-123403@example.com.ics' from 'user1/pythoncaldav-test': 'list' object has no attribute 'lower'
```

Here is the HTTP communication:

```
REPORT http://localhost:5232/user1/pythoncaldav-test/
Depth: 1
Content-Type: application/xml; charset="utf-8"

<?xml version='1.0' encoding='utf-8'?>
<C:calendar-query xmlns:D="DAV:" xmlns:C="urn:ietf:params:xml:ns:caldav">
  <D:prop>
    <C:calendar-data/>
  </D:prop>
  <C:filter>
    <C:comp-filter name="VCALENDAR">
      <C:comp-filter name="VEVENT">
        <C:prop-filter name="CATEGORIES">
          <C:text-match collation="i;octet">PERSONAL</C:text-match>
        </C:prop-filter>
      </C:comp-filter>
    </C:comp-filter>
  </C:filter>
</C:calendar-query>
<====
500 Internal Server ErrorDate: Fri, 27 Jan 2023 11:14:01 GMT
Server: WSGIServer/0.2 CPython/3.10.9
Content-Type: text/plain
Content-Length: 59

A server error occurred.  Please contact the administrator.
```

And here is the ical I'm expecting to get in return from the calendar:

```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//Example Corp.//CalDAV Client//EN
BEGIN:VEVENT
UID:19970901T130000Z-123403@example.com
DTSTAMP:19970901T130000Z
DTSTART;VALUE=DATE:19971102
SUMMARY:Our Blissful Anniversary
TRANSP:TRANSPARENT
CLASS:CONFIDENTIAL
CATEGORIES:ANNIVERSARY,PERSONAL,SPECIAL OCCASION
RRULE:FREQ=YEARLY
END:VEVENT
END:VCALENDAR
```

**Comment by tobixen** [comment]
Seems like this issue has been fixed at some point.  I thought I was running my tests towards the master branch of Radicale, but I probably messed it up, because I'm not able to reproduce the problem anymore.

**Comment by tobixen** [comment]
Sorry - I'm apparently quite confused - I actually wrote up a quickfix months ago and left it uncommitted on my laptop.  I can also see that this issue apparntly has been fixed more properly in 2a644e753d2740daa375cad115445c87aad02ef3.

This issue should anyway be closed as it's a duplicate.
